akka-persistence-tck/src/main/scala/akka/persistence/journal/JournalSpec.scala

```diff
@@ -57,12 +57,12 @@ abstract class JournalSpec(config: Config) extends PluginSpec(config) {
 
   def writeMessages(fromSnr: Int, toSnr: Int, pid: String, sender: ActorRef, writerUuid: String): Unit = {
     val msgs =
-      if (supportsAtomicPersistAllOfSeveralEvents)
-        (fromSnr to toSnr).map { i ⇒
-          AtomicWrite(PersistentRepr(payload = s"a-$i", sequenceNr = i, persistenceId = pid, sender = sender,
-            writerUuid = writerUuid))
-        }
-      else
+      if (supportsAtomicPersistAllOfSeveralEvents) {
+        List(AtomicWrite((fromSnr to toSnr) map { i ⇒
+          PersistentRepr(payload = s"a-$i", sequenceNr = i, persistenceId = pid, sender = sender,
+            writerUuid = writerUuid)
+        }))
+      } else {
         (fromSnr to toSnr - 1).map { i ⇒
           if (i == toSnr - 1)
             AtomicWrite(List(
```

Comment:
Yes, this is still wrong;
The first case should be `List(AtomicWrite(1,2,3,...))` the second case should be `List(AtomicWrite(1), AtomicWrite(2), AtomicWrite(3), ...)`.

The intent is to allow journals to implement handling of `AtomicWrite(1,2,3)` by throwing since there's no way it can guarantee it. Such journal should still pass, while it should explicitly document it is not able to provide atomic writes of more than 1 event.

// Yes, it could "store them as one event internally" but we're want to leave this up to the implementation if it does some hack, or says it can not handle this.